aten/src/ATen/native/LossNLL.cpp

```diff
@@ -455,19 +455,74 @@ Tensor nll_loss_backward_cpu(
   return grad_input;
 }
 
+Tensor cross_entropy_loss_prob_target(
+    const Tensor& self,
+    const Tensor& target,
+    const Tensor& weight,
+    int64_t reduction) {
+  const auto n_classes = self.size(1);
+  TORCH_CHECK(
+      !weight.defined() || (weight.dim() == 1 && weight.numel() == n_classes),
+      "weight tensor should be defined either for all ",
+      n_classes,
+      " classes or no classes"
+      " but got weight tensor of shape: ",
+      weight.sizes());
+
+  Tensor ret;
+  Tensor input = at::log_softmax(
+      self, 1, optTypeMetaToScalarType(self.options().dtype_opt()));
```

Comment:
optTypeMetaToScalarType(self.options().dtype_opt()) -> `self.scalar_type` should work